### ISSUE TITLE
Add actions and reducer to login and logout the user

### ISSUE THREAD

**Issue by Odrakir** [toggl/watchos-app#9]
We will probably need the following actions:

- login (with email and password for now)
- setCurrentUser (which will include the user as payload)
- logout

The login action will return an effect that will set the current user when successful.

Also handle errors. Probably adding them to the state so we bind them in the UI.

Finally add tests. The reducer is a pure function so it should be easy to add unit tests for it.

So, to recap:
- [x] Add user actions
- [x] Add the user reducer
- [ ] Watch out for errors, add necessary state
- [ ] Add unit tests on the reducer
